NuRadioMC/EvtGen/generator.py

```diff
@@ -1364,7 +1364,7 @@ def J(E):
         mask_leptons = mask_leptons & mask_theta & mask_phi
 
         E_all_leptons = (1 - data_sets["inelasticity"]) * data_sets["energies"]
-        lepton_codes = data_sets["flavors"]
+        lepton_codes = data_sets["flavors"][:]
```

Comment:
what is this doing? 
If you want to be sure to copy something I'd suggest to use
```python
import copy
lepton_codes = copy.copy(data_sets["flavors"])
```

Reply:
This is the native pythonic way of copying an array-like structure by value instead of reference. I can also use numpy.copy

Reply:
So, to elaborate, the brackets take a slice and return the slice by value. A slice of the type [:] is equivalent to [None:None], which is the same thing as returning a slice that encompasses the whole array.

Reply:
From https://stackoverflow.com/questions/2612802/how-to-clone-or-copy-a-list:
"Alex Martelli's opinion (at least back in 2007) about this is, that it is a weird syntax and it does not make sense to use it ever. ;) (In his opinion, the next one is more readable)."
I agree with that, it makes the code more obscure. Rather implement it explicitly using the copy module.

Reply:
Changed.